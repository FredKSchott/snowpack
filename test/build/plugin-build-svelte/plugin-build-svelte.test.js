const fs = require('fs');
const path = require('path');
const {setupBuildTest} = require('../../test-utils');

const cwd = path.join(__dirname, 'build');

describe('@snowpack/plugin-build-svelte', () => {
  beforeAll(() => {
    setupBuildTest(__dirname);
  });

  it('builds source svelte files as expected', () => {
    const jsLoc = path.join(cwd, '_dist_', 'index.svelte.js');
    expect(fs.existsSync(jsLoc)).toBe(true); // file exists
    expect(fs.readFileSync(jsLoc, 'utf-8')).toContain(`import { refresh, comment, camera } from "../_snowpack/pkg/svelte-awesome/icons.js";`); // file has expected imports
    expect(fs.readFileSync(jsLoc, 'utf-8')).toContain(`import './index.svelte.css.proxy.js';`); // file has expected imports

    const cssLoc = path.join(cwd, '_dist_', 'index.svelte.css.proxy.js');
    expect(fs.existsSync(cssLoc)).toBe(true); // file exists
  });

  it('builds package svelte files as expected', () => {
    expect(fs.existsSync(path.join(cwd, '_snowpack', 'pkg', 'svelte-awesome.js'))).toBe(true); // import exists
    expect(fs.readFileSync(path.join(cwd, '_snowpack', 'pkg', 'svelte-awesome', 'icons.js'), 'utf-8')).toMatchSnapshot('svelte-awesome treeshaking'); // import exists, and was tree-shaken
  });
});
