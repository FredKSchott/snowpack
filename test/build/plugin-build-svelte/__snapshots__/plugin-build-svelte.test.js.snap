// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@snowpack/plugin-build-svelte builds package svelte files as expected: svelte-awesome treeshaking 1`] = `
"var refresh = { refresh: { width: 1536, height: 1792, paths: [{ d: 'M1511 1056q0 5-1 7-64 268-268 434.5t-478 166.5q-146 0-282.5-55t-243.5-157l-129 129q-19 19-45 19t-45-19-19-45v-448q0-26 19-45t45-19h448q26 0 45 19t19 45-19 45l-137 137q71 66 161 102t187 36q134 0 250-65t186-179q11-17 53-117 8-23 30-23h192q13 0 22.5 9.5t9.5 22.5zM1536 256v448q0 26-19 45t-45 19h-448q-26 0-45-19t-19-45 19-45l138-138q-148-137-349-137-134 0-250 65t-186 179q-11 17-53 117-8 23-30 23h-199q-13 0-22.5-9.5t-9.5-22.5v-7q65-268 270-434.5t480-166.5q146 0 284 55.5t245 156.5l130-129q19-19 45-19t45 19 19 45z' }] } };

var camera = { camera: { width: 1920, height: 1792, paths: [{ d: 'M960 672q119 0 203.5 84.5t84.5 203.5-84.5 203.5-203.5 84.5-203.5-84.5-84.5-203.5 84.5-203.5 203.5-84.5zM1664 256q106 0 181 75t75 181v896q0 106-75 181t-181 75h-1408q-106 0-181-75t-75-181v-896q0-106 75-181t181-75h224l51-136q19-49 69.5-84.5t103.5-35.5h512q53 0 103.5 35.5t69.5 84.5l51 136h224zM960 1408q185 0 316.5-131.5t131.5-316.5-131.5-316.5-316.5-131.5-316.5 131.5-131.5 316.5 131.5 316.5 316.5 131.5z' }] } };

var comment = { comment: { width: 1792, height: 1792, paths: [{ d: 'M1792 896q0 174-120 321.5t-326 233-450 85.5q-70 0-145-8-198 175-460 242-49 14-114 22-17 2-30.5-9t-17.5-29v-1q-3-4-0.5-12t2-10 4.5-9.5l6-9t7-8.5 8-9q7-8 31-34.5t34.5-38 31-39.5 32.5-51 27-59 26-76q-157-89-247.5-220t-90.5-281q0-130 71-248.5t191-204.5 286-136.5 348-50.5q244 0 450 85.5t326 233 120 321.5z' }] } };

export { camera, comment, refresh };
"
`;
